aspnetcore/security/authentication/facebook-logins.md

```diff
@@ -0,0 +1,122 @@
+
+![image](sociallogins/_static/FBProductSetup.png)
+
+* You will be presented with the **Client OAuth Settings** page with some defaults already set:
+
+![image](sociallogins/_static/FBOAuthSetup.png)
+
+* Enter your base URI with *signin-facebook* appended into the **Valid OAuth Redirect URIs** field. For example, https://localhost:44320/**signin-facebook**.
+
+> [!NOTE]
+> You don't need to configure **signin-facebook** as a route in your app. The ASP.NET Core team's implementation of the OAuth flow will create a temporary socket (called a *backchannel*) that listens at this route just for the duration of the OAuth flow.
+
+* Make a note of your *App ID* and your *App Secret* before dismissing this page so that you can add both into your ASP.NET Core app later in this tutorial.
+
+* Tap **Save Changes** to complete the new application configuration.
+
+## Storing Facebook AppId and AppSecret
+
+Link sensitive settings like Facebook *AppID* and *Secret* to your application configuration by using the [Secret Manager tool](../app-secrets.md) instead of storing them in your configuration file directly, as described in the [previous section](sociallogins.md). Execute the following in your project working directory:
+
+* Set the Facebook AppId
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppId <app-Id>
+     ````
+
+* Set the Facebook AppSecret
+
+  <!-- literal_block {"ids": [], "xml:space": "preserve"} -->
+
+  ````
+  dotnet user-secrets set Authentication:Facebook:AppSecret <app-secret>
+     ````
+
+The following code reads the configuration values stored by the [Secret Manager](../app-secrets.md#security-app-secrets):
+
+[!code-csharp[Main](../../common/samples/WebApplication1/Startup.cs?highlight=11&range=20-36)]
+
+## Enable Facebook middleware
+
+> [!NOTE]
+> You will need to use NuGet to install the [Microsoft.AspNetCore.Authentication.Facebook](https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.Facebook/1.1.0-preview1-final) package if it hasn't already been installed. Alternatively, execute the following in your project directory:
+>
+> `dotnet install Microsoft.AspNetCore.Authentication.Facebook`
+
+Add the Facebook middleware in the `Configure` method in `Startup.cs`:
+
+[!code-csharp[Main](./sociallogins/sample/Startup.cs?highlight=21,22,23,24,25&range=58-110)]
+
+## Login with Facebook
+
+Run your application and click Login. You will see an option for Facebook.
+
+![image](sociallogins/_static/FBLogin1.PNG)
+
+When you click on Facebook, you will be redirected to Facebook for authentication.
+
+![image](sociallogins/_static/FBLogin2.PNG)
+
+Once you enter your Facebook credentials, then you will be redirected back to the Web site where you can set your email.
+
+You are now logged in using your Facebook credentials:
+
+![image](sociallogins/_static/Done.PNG)
+
+## Optionally set password
```

Comment:
For now, taking out of provider-specific pages and moving to index page for social logins.